cmd/internal/kube/inject.go

```diff
@@ -0,0 +1,229 @@
+package kube
+
+import (
+	"bytes"
+	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
+	"github.com/akitasoftware/akita-cli/cmd/internal/kube/injector"
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	injectFileNameFlag string
+	injectOutputFlag   string
+	projectNameFlag    string
+	secretInjectFlag   string
+)
+
+var injectCmd = &cobra.Command{
+	Use:   "inject",
+	Short: "Inject Akita into a Kubernetes deployment",
+	Long:  "Inject Akita into a Kubernetes deployment or set of deployments and output the result to stdout or a file",
+	RunE: func(_ *cobra.Command, args []string) error {
+		secretOpts := resolveSecretGenerationOptions(secretInjectFlag)
+
+		// If the user specified a secret file path, but did not specify an output file, return an error
```

Comment:
I don't think it fits into the intended usecase of the command, but I guess we could allow it?

Users would likely create both the Deployment and Secret resources at the same time by either applying everything as a file:
```
akita kube inject -s="super-secrety.yml" -f in.yml -o out.yml
# This part could also be handled by some other tooling
kubectl apply -f out.yml -f super-secrety.yml
```
Or, they'd just apply the output at once via pipe:
```
akita kube inject -s -f in.yml | kubectl apply -f -
```

Since we've also created a separate command for generating and applying Secrets in isolation (https://github.com/akitasoftware/akita-cli/pull/202), I'd prefer that we leave this as-is to be more explicit about the intended usecase of the command. My main worry is we'll have users doing something like this and not fully understanding why their applies are failing:
```
akita kube inject -s="fudged-it.yml" -f in.yml | kubectl apply -f -
```

Reply:
Having said that, this could absolutely use some more documentation for end-users and for us, so I'll update it!

Reply:
I've added better documentation in [b6ad9d1](https://github.com/akitasoftware/akita-cli/pull/206/commits/b6ad9d1790c09b2a8e56efba83fc04ac6cad0257), but I'll leave this open if you have a strong preference on removing this check

Reply:
This is fine, but we should mention this requirement in the help message for `--secret`, as I've suggested in my follow-up review.